applications/FluidDynamicsApplication/custom_elements/embedded_fluid_element_discontinuous.cpp

```diff
@@ -0,0 +1,975 @@
+        const BoundedMatrix<double, LocalSize, LocalSize> aux_matrix_BtransAtransPtanACB = prod(aux_matrix_BtransAtransPtan, aux_matrix_ACB);
+
+        // Contribution coming from the traction vector tangencial component
+        noalias(aux_LHS_1) -= adjoint_consistency*nitsche_coefs.first*weight*aux_matrix_BtransAtransPtanACB;
+
+        // Contribution coming from the shear force generated by the velocity jump
+        noalias(aux_LHS_2) -= adjoint_consistency*nitsche_coefs.second*weight*prod(aux_matrix_BtransAtransPtan, N_mat);
+    }
+
+    // Compute negative side LHS contribution
+    const unsigned int number_of_negative_interface_integration_points = rData.NegativeInterfaceWeights.size();
+    for (unsigned int g = 0; g < number_of_negative_interface_integration_points; ++g){
+        // Get the Gauss pt. data
+        const double weight = rData.NegativeInterfaceWeights[g];
+        const auto aux_N = row(rData.NegativeInterfaceN, g);
+        const BoundedMatrix<double, NumNodes, Dim> aux_DN_DX = rData.NegativeInterfaceDNDX[g];
+        const auto &aux_unit_normal = rData.NegativeInterfaceUnitNormals[g];
+
+        // Set the shape functions auxiliar matrices
+        BoundedMatrix<double, Dim, LocalSize> N_mat = ZeroMatrix(Dim, LocalSize);
+        for (unsigned int i = 0; i < NumNodes; ++i){
+            for (unsigned int comp = 0; comp < Dim; ++comp){
+                N_mat(comp, i*BlockSize + comp) = aux_N(i);
+            }
+        }
+
+        // Set the current Gauss pt. strain matrix
+        BoundedMatrix<double, StrainSize, LocalSize> B_matrix = ZeroMatrix(StrainSize, LocalSize);
+        FluidElementUtilities<NumNodes>::GetStrainMatrix(aux_DN_DX, B_matrix);
+
+        // Set the tangential projection matrix (I - n x n)
+        BoundedMatrix<double, Dim, Dim> tang_proj_matrix;
+        FluidElementUtilities<NumNodes>::SetTangentialProjectionMatrix(aux_unit_normal, tang_proj_matrix);
+
+        // Get the normal projection matrix in Voigt notation
+        BoundedMatrix<double, Dim, StrainSize> voigt_normal_proj_matrix = ZeroMatrix(Dim, StrainSize);
+        FluidElementUtilities<NumNodes>::VoigtTransformForProduct(aux_unit_normal, voigt_normal_proj_matrix);
+
+        // Compute some Gauss pt. auxiliar matrices
+        const BoundedMatrix<double, LocalSize, Dim> aux_matrix_BtransAtrans = prod(trans(B_matrix), trans(voigt_normal_proj_matrix));
+        const BoundedMatrix<double, LocalSize, Dim> aux_matrix_BtransAtransPtan = prod(aux_matrix_BtransAtrans, tang_proj_matrix);
+        const BoundedMatrix<double, StrainSize, LocalSize> aux_matrix_CB = prod(rData.C, B_matrix);
+        const BoundedMatrix<double, Dim, LocalSize> aux_matrix_ACB = prod(voigt_normal_proj_matrix, aux_matrix_CB);
+        const BoundedMatrix<double, LocalSize, LocalSize> aux_matrix_BtransAtransPtanACB = prod(aux_matrix_BtransAtransPtan, aux_matrix_ACB);
```

Comment:
@adityaghantasala did u finally make that famous optimized function for the calculation of Ttrans A T ?

Reply:
ping @adityaghantasala